src/nrepl/jack-in.ts

```diff
@@ -53,10 +53,10 @@ function getGlobalJackInEnv() {
     };
 }
 
-async function executeJackInTask(
+function executeJackInTask(
     terminalOptions: JackInTerminalOptions,
     connectSequence: ReplConnectSequence,
-    cb?: Function
+    cb?: () => unknown
```

Comment:
Again, not great to use `Function`. As an aside, using `unknown` is typically better than using `any`, since using `any` defeats the purpose of the type system. So either you don't know what it is, in which case use unknown, or you know it should have specific fields or be callable, in which case create a type for it.